tests/unit/s2n_client_key_share_extension_pq_test.c

```diff
@@ -0,0 +1,1069 @@
+
+                        uint16_t expected_hybrid_share_size =
+                                S2N_SIZE_OF_KEY_SHARE_SIZE
+                                + negotiated_kem_group->curve->share_size
+                                + S2N_SIZE_OF_KEY_SHARE_SIZE
+                                + negotiated_kem_group->kem->public_key_length;
+                        uint16_t sent_hybrid_share_size;
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &sent_hybrid_share_size));
+                        EXPECT_EQUAL(sent_hybrid_share_size, expected_hybrid_share_size);
+
+                        uint16_t hybrid_ecc_share_size;
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &hybrid_ecc_share_size));
+                        EXPECT_EQUAL(hybrid_ecc_share_size, negotiated_kem_group->curve->share_size);
+                        EXPECT_SUCCESS(s2n_stuffer_skip_read(&key_share_extension, hybrid_ecc_share_size));
+
+                        uint16_t hybrid_pq_share_size;
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &hybrid_pq_share_size));
+                        EXPECT_EQUAL(hybrid_pq_share_size, negotiated_kem_group->kem->public_key_length);
+                        EXPECT_SUCCESS(s2n_stuffer_skip_read(&key_share_extension, hybrid_pq_share_size));
+
+                        /* If all the sizes/bytes were correctly written, there should be nothing left over */
+                        EXPECT_EQUAL(s2n_stuffer_data_available(&key_share_extension), 0);
+
+                        EXPECT_SUCCESS(s2n_connection_free(conn));
+                    }
+                }
+
+                /* Test failure when server chooses a KEM group that is not in the client's preferences */
+                {
+                    struct s2n_connection *conn;
+                    EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+                    conn->security_policy_override = &security_policy_sike;
+                    conn->actual_protocol_version = S2N_TLS13;
+                    conn->handshake.handshake_type = HELLO_RETRY_REQUEST;
+                    conn->handshake.message_number = HELLO_RETRY_MSG_NO;
+                    conn->actual_protocol_version_established = 1;
+
+                    conn->secure.server_kem_group_params.kem_group = &s2n_secp256r1_bike1_l1_r2;
+
+                    DEFER_CLEANUP(struct s2n_stuffer key_share_extension = {0}, s2n_stuffer_free);
+                    EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&key_share_extension, 4096));
+                    EXPECT_FAILURE_WITH_ERRNO(s2n_client_key_share_extension.send(conn, &key_share_extension), S2N_ERR_INVALID_HELLO_RETRY);
+
+                    EXPECT_SUCCESS(s2n_connection_free(conn));
+                }
+            }
+        }
+        /* Tests for s2n_client_key_share_extension.recv */
```

Comment:
Missing newline (for a bunch of tests -- the copied half of the file adds a newline, the new tests do not)